src/main/java/baritone/cache/FasterWorldScanner.java

```diff
@@ -191,7 +192,29 @@ private void visitSection(BlockOptionalMetaLookup lookup, LevelChunkSection sect
             return;
         }
 
-        boolean[] isInFilter = getIncludedFilterIndices(lookup, ((IPalettedContainer<BlockState>) sectionContainer).getPalette());
+        int yOffset = section.bottomBlockY();
+        Palette<BlockState> palette = ((IPalettedContainer<BlockState>) sectionContainer).getPalette();
+
+        if (palette instanceof SingleValuePalette) {
+            // single value palette doesn't have any data
+            if (lookup.has(palette.valueFor(0))) {
+                // TODO this is 4k hits, maybe don't return all of them?
+                for (int x = 0; x < 16; ++x) {
+                    for (int y = 0; y < 16; ++y) {
+                        for (int z = 0; z < 16; ++z) {
+                            blocks.add(new BlockPos(
```

Comment:
Does it use SingleValuePalette if the section is all entirely the same block?

Reply:
Yes